### PR TITLE
installation of GridTools C++ sources without setup.py

### PR DESCRIPTION
**Pull request by egparedes**
Move the installation of GridTools C++ sources out of `setup.py` script

### CODE REVIEW

File: src/gt4py/gt_src_manager.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    parser = argparse.ArgumentParser(description="Manage GridTools C++ code sources")
+    parser.add_argument("command", choices=["install", "remove", "status"])
+    args = parser.parse_args()
+
+    if args.command == "install":
+        install_gt_sources()
+        _print_status()
+    elif args.command == "remove":
+        remove_gt_sources()
+        _print_status()
```
Comment (gronerl) [inline src/gt4py/gt_src_manager.py]: this prints `GridTools are NOT properly installed in the current environment!` also when removing which looks a bit errory, maybe print something like "GridTools successfully removed" ?

File: src/gt4py/gt_src_manager.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    if os.path.exists(target_path):
+        rm_cmd = f"rm -Rf {target_path}"
+        print(f"Deleting sources...\n$ {rm_cmd}")
+        subprocess.run(rm_cmd.split())
+
+
+def has_gt_sources():
+    return os.path.isfile(
+        os.path.join(gt_config.GT_INCLUDE_PATH, "gridtools", "common", "defs.hpp")
```
Comment (gronerl) [inline src/gt4py/gt_src_manager.py]: could also check that the sources are in the right version (and potentially check out the right one corresponding to the checked out gt4py commit)  - but this is a feature orthogonal to this refactoring change

File: src/gt4py/backend/base_gt_backend.py [hunk 3]
```diff
@@ -498,15 +498,13 @@ def validate_cache_info(cls, stencil_id, cache_info):
...
 
         # Generate the Python binary extension (checking if GridTools sources are installed)
-        if not os.path.isfile(
-            os.path.join(gt_config.GT_INCLUDE_PATH, "gridtools", "common", "defs.hpp")
-        ):
-            raise RuntimeError(
-                "Missing GridTools sources. Run 'python setup.py install_gt_sources'."
-            )
+        if not gt_src_manager.has_gt_sources() and not gt_src_manager.install_gt_sources():
+            raise RuntimeError("Missing GridTools sources.")
```
Comment (gronerl) [inline src/gt4py/backend/base_gt_backend.py]: Why remove the hint `Run 'python -m gt4py.gt_src_manager install'` ?

File: src/gt4py/gt_src_manager.py (hunk 1 above)
Comment (egparedes) [reply src/gt4py/gt_src_manager.py]: Done

File: src/gt4py/gt_src_manager.py (hunk 2 above)
Comment (egparedes) [reply src/gt4py/gt_src_manager.py]: I agree that could be a useful in the future, but it is out of scope for this PR.

File: src/gt4py/backend/base_gt_backend.py (hunk 3 above)
Comment (egparedes) [reply src/gt4py/backend/base_gt_backend.py]: If the code gets to this point, the hint should have been already printed in the failed `gt_src_manager.install_gt_sources` function

**Pull request merged**